src/coreclr/src/vm/methodtable.h

```diff
@@ -468,15 +468,16 @@ struct MethodTableWriteableData
                        MethodTableWriteableData::enum_flag_Unrestored);
     }
 
-    void SetIsFullyLoadedForBuildMethodTable()
+    void SetIsRestoredForBuildArrayMethodTable()
     {
         LIMITED_METHOD_CONTRACT;
 
         // Used only during method table initialization - no need for logging or Interlocked Exchange.
-        m_dwFlags &= ~(MethodTableWriteableData::enum_flag_UnrestoredTypeKey |
-                       MethodTableWriteableData::enum_flag_Unrestored |
-                       MethodTableWriteableData::enum_flag_IsNotFullyLoaded |
-                       MethodTableWriteableData::enum_flag_HasApproxParent);
+        SetIsRestoredForBuildMethodTable();
```

Comment:
What motivated this change?

Reply:
To match the initial load level of ArrayTypeDesc.
Otherwise the array MT is always fully loaded, even if when the element type may be not fully loaded.
I did not see such cases, but it seems possible.

Reply:
This will also force some checks that are done when moving to fully loaded state - like that the type is not infinitely recursive.

---
In reply to: [362986502](https://github.com/dotnet/runtime/pull/1201#discussion_r362986502) [](ancestors = 362986502)